File: apps/backend/src/FinanceManager.Domain/Entities/ContaPagar.cs
using System;

namespace FinanceManager.Domain.Entities
{
    public class ContaPagar
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public string Descricao { get; set; }
        public decimal Valor { get; set; }
        public DateTime DataVencimento { get; set; }
        public string Status { get; set; } // Ex: "Pendente", "Pago", "Atrasado"
        public string Categoria { get; set; } // Ex: "Alimentação", "Transporte", "Saúde"
        public DateTime CreatedAt { get; set; }
    }


    }   

File: apps/frontend/src/types/ContaPagar.ts
export interface CreateContaPagarRequest {
    userId: string;
    descricao: string;
    valor: number;
    dataVencimento: string; 
    categoria: string;
}

export interface UpdateContaPagarRequest {
    id: string;
    descricao: string;
    valor: number;
    dataVencimento: string; 
    categoria: string;
    status: 'Pendente' | 'Pago';
}

export interface ContaPagarDto {
    id: string;
    descricao: string;
    valor: number;
    dataVencimento: string; 
    categoria: string;
    status: 'Pendente' | 'Pago';
    createdAt: string; 
}

export interface Paginated<T> {
    items: T[];
}
